### PR TITLE
report: update error icons

### LINKED ISSUE #9820: Runtime Error Design Update

### PR DESCRIPTION
**Pull request by tannerdolby**
**Summary**

Fixes #9820 

Updates runtime error icons to use the SVGs provided in https://github.com/GoogleChrome/lighthouse/issues/9820#issuecomment-541796558

Screen Shot 2023-05-17 at 11 30 28 PM

**Testing**
- run `yarn build-sample-reports`
- view the .cdt.error/index.html or .psi.error/index.html sample reports
- see the updated runtime error icon

**Related Issues/PRs**

https://github.com/GoogleChrome/lighthouse/pull/9272
https://github.com/GoogleChrome/lighthouse/issues/9968#issuecomment-675704398

### CODE REVIEW

File: report/assets/styles.css [hunk 1]
```diff
@@ -167,6 +167,9 @@
...
   --fail-icon-url: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 48 48"><title>warn</title><path fill="%23C7221F" d="M2 42h44L24 4 2 42zm24-6h-4v-4h4v4zm0-8h-4v-8h4v8z"/></svg>');
+  --error-icon-sm-url: url('data:image/svg+xml;utf8,<svg width="18" height="28" viewBox="0 0 18 28" xmlns="http://www.w3.org/2000/svg"><title>lighthouse error</title><path d="M7.5 22H10.5V19H7.5V22Z" fill="%23FF4E42"/><path d="M7.5 16H10.5V7H7.5V16Z" fill="%23FF4E42"/></svg>');
+  --error-icon-md-url: url('data:image/svg+xml;utf8,<svg width="36" height="36" viewBox="0 0 36 36" xmlns="http://www.w3.org/2000/svg"><title>lighthouse error</title><path d="M17 26H20V23H17V26Z" fill="%23FF4E42"/><path d="M17 20H20V11H17V20Z" fill="%23FF4E42"/></svg>');
+  --error-icon-lg-url: url('data:image/svg+xml;utf8,<svg width="120" height="120" viewBox="0 0 120 120" xmlns="http://www.w3.org/2000/svg"><title>lighthouse error</title><path d="M56 80H64V72.0001H56V80Z" fill="%23FF4E42"/><path d="M56 64H64V41H56V64Z" fill="%23FF4E42"/></svg>');
```
Comment (adamraine) [inline report/assets/styles.css:172]: svgs should scale just fine on their own, why do we need 3 different sizes?

**Comment by adamraine** [comment]
Looks good this is a great start! It would be nice to have a single definition for this svg rather than four different ones (3 sizes in CSS, and 1 created in JS).

Also, noticed the icon was slightly off center in audit lists:

Screenshot 2023-05-18 at 10 37 47 AM

File: report/assets/styles.css (hunk 1 above)
Comment (tannerdolby) [reply report/assets/styles.css:172]: agreed, the svgs will scale fine on their own. We don't need the 3 different sizes so I'll go ahead and update this.

**Comment by tannerdolby** [comment]
@adamraine Thanks for the review! I've updated the svg definitions and now there is only one definition in CSS. I did some tinkering and found a way to get the positioning right using only background-images.

I think the icon being off center in the audit list was partly due to the svg definition I originally had, it should be updated now and appear centered in the row on Chrome (Version 113.0.5672.126) and Firefox (109.0.1). Let me know if it still appears off and I'll try to reproduce the alignment issue to fix it.


Screen Shot 2023-05-18 at 8 12 06 PM

File: report/assets/styles.css [hunk 2]
```diff
@@ -561,6 +562,38 @@
...
+  top: 4px;
+}
+
+.lh-gauge__wrapper--fail:not([href]) .lh-gauge--error {
+  background-image: var(--error-icon-url);
+  background-repeat: no-repeat;
+  height: 120px;
+  left: 139px;
+  top: -5px;
+}
```
Comment (adamraine) [inline report/assets/styles.css:595]: We can avoid using exact pixel values with `background-position: center` and `var(--score-container-padding)`

```suggestion
.lh-gauge__wrapper--fail .lh-gauge--error {
  background-image: var(--error-icon-url);
  background-repeat: no-repeat;
  background-position: center;
  top: var(--score-container-padding);
}
```

File: report/assets/styles.css [hunk 3]
```diff
@@ -561,6 +562,38 @@
...
+.lh-audit--error .lh-audit__score-icon,
+.lh-metric--error .lh-metric__icon {
+  background-image: var(--error-icon-url);
+  background-repeat: no-repeat;
+  width: calc(var(--score-icon-size) * 2.4);
+  height: calc(var(--score-icon-size) * 2.4);
+  transform: translateX(-6px);
+  margin-right: 0;
+  border: none;
+}
```
Comment (adamraine) [inline report/assets/styles.css:574]:
```suggestion
.lh-audit--error .lh-audit__score-icon,
.lh-metric--error .lh-metric__icon {
  background-image: var(--error-icon-url);
  background-repeat: no-repeat;
  width: calc(var(--score-icon-size) * 2.4);
  height: calc(var(--score-icon-size) * 2.4);
  background-position: center;
  margin-right: 0;
  border: none;
}
```

File: report/assets/styles.css (hunk 2 above)
Comment (tannerdolby) [reply report/assets/styles.css:595]: Good callout, will update.

File: report/assets/styles.css (hunk 3 above)
Comment (tannerdolby) [reply report/assets/styles.css]: Updated!

File: report/assets/styles.css [hunk 4]
```diff
@@ -561,6 +562,36 @@
...
+  height: calc(var(--score-icon-size) * 3.25);
+  background-position: center;
+  top: calc(var(--score-container-padding) / 2);
+}
+
+.lh-gauge__wrapper--fail:not([href]) .lh-gauge--error {
+  background-image: var(--error-icon-url);
+  background-repeat: no-repeat;
+  background-position: center;
+}
```
Comment (adamraine) [inline report/assets/styles.css:593]: This rule doesn't actually change anything now and can safely be deleted

File: report/assets/styles.css [hunk 5]
```diff
@@ -561,6 +562,36 @@
...
+  background-repeat: no-repeat;
+  background-position: center;
+  top: var(--score-container-padding);
+}
+
+.lh-sticky-header .lh-gauge__wrapper--fail .lh-gauge--error {
+  height: calc(var(--score-icon-size) * 3.25);
+  background-position: center;
+  top: calc(var(--score-container-padding) / 2);
+}
```
Comment (adamraine) [inline report/assets/styles.css:587]: If we remove this rule, the "!" just gets smaller. I would prefer to do that and keep the code concise.

File: report/assets/styles.css (hunk 4 above)
Comment (tannerdolby) [reply report/assets/styles.css:593]: sounds good, will remove it.

File: report/assets/styles.css (hunk 5 above)
Comment (tannerdolby) [reply report/assets/styles.css:587]: Cool, just removed the rule. Like you mentioned the "!" gets smaller but it still looks good.

**Review by adamraine — APPROVED** [review]
This looks great! I made some changes to scale up the svg to fit it's view box, then scale it down by 50% when it's in a score gauge. This allows us to inherit a lot of the same CSS used to render the normal score icons in audits and metrics.

**Comment by tannerdolby** [comment]
Awesome! I just checked out your changes and everything looks great. That was a smart idea to scale the svg to allow for us to inherit a lot of the same CSS.

**Pull request merged**